hilti/runtime/src/types/bytes.cc

```diff
@@ -193,6 +193,9 @@ uint64_t Bytes::toUInt(ByteOrder byte_order) const {
         case ByteOrder::Big: break;
     }
 
+    if ( isEmpty() )
+        throw RuntimeError("not enough bytes for conversion to integer");
```

Comment:
Let's throw the same error we throw for `atoi_n` unless there's a good reason to deviate.

```suggestion
        throw InvalidArgument("cannot decode from empty range");
```

Reply:
This was aligning to the RuntimeError just below (which seems in the same category). I'm also not quite sure I like the "decode" for interpreting 4 bytes as int (rather than decoding a ascii string to an integer).

If you confirm you prefer both, I'll change.

Reply:
> InvalidArgument

This one is documented as "Exception flagging invalid arguments passed to a function." - so it doesn't quite fit. I could see `InvalidValue` being more suitable, but would want to replace the below `throw` then, too.

Reply:
`InvalidValue` would sound good to me for both (and then actually also `InvalidArgumet` for the current `RuntimeError` thrown at the beginning for `Undef`).

That said, this all isn't clear cut because we aren't very consistent across the runtime library in what exactly to throw when. The important part is that both `InvalidArgument` and `InvalidValue` do derive from `RuntimeError`, so we get that  either way. `RuntimeError` is what Spicy's error handling keys on, so that's important. The specific derived class then just adds a bit of semantic value that however doesn't make a functional difference.